packages/synapse-react-client/test/utils/hooks/useMutuallyExclusiveState.test.ts

```diff
@@ -0,0 +1,60 @@
+import { act, renderHook } from '@testing-library/react'
+import useMutuallyExclusiveState from '../../../src/utils/hooks/useMutuallyExclusiveState'
+
+describe('useMutuallyExclusiveState', () => {
+  it('throws error if both initial state values are true', () => {
+    const consoleErrorSpy = jest
+      .spyOn(console, 'error')
+      .mockImplementation(() => {})
+
+    expect(() =>
+      renderHook(() => useMutuallyExclusiveState(true, true)),
+    ).toThrowError()
+
+    consoleErrorSpy.mockRestore()
+  })
+  it('allows both values to be false', () => {
+    const hookReturn = renderHook(() => useMutuallyExclusiveState(false, false))
```

Comment:
Rather the return object has explicitly named values rather than accessing via index.

```
const [
    showValue1,
    setValue1,
    showValue2,
    setValue2,
  ] = useMutuallyExclusiveState(false, false)
```

Reply:
Decided to return an array to match the API for `useState`. I think the reason they do that is because 
1. you're always going to use every single return value
2. the variable name will always be overwritten with a better semantic name